video/enhanced_samples_generator.h

```diff
@@ -83,7 +83,10 @@ int EnhancedSamplesGenerator::getSample() {
 }
 
 int EnhancedSamplesGenerator::getDuration(uint16_t frequency) {
-	return _sample.expired() ? 0 : _sample.lock()->getDuration() / calculateSamplerate(frequency);
+	// TODO this will produce an incorrect duration if the sample rate for the channel has been
+	// adjusted to differ from the underlying audio system sample rate
+	// At this point it's not clear how to resolve this, so we'll assume it hasn't been adjusted
+	return _sample.expired() ? 0 : (_sample.lock()->getSize() * 1000 / sampleRate()) / calculateSamplerate(frequency);
```

Comment:
making this work if the channel alone has had its sample rate adjusted will be tricky, and likely involve a reasonably complex change to the audio system

given that this is here for the convenience of specifying a zero duration to play an entire sample, it feels reasonable to say that zero duration will only work properly if the channel is running at the same sample rate as the underlying audio system